src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java

```diff
@@ -177,6 +188,33 @@ public void insertReferencedSamples(SubmissionEnvelope submissionEnvelope) {
         submissionEnvelope.getSupportingSamples().addAll((Set<? extends Sample>) refLookupService.lookupRefs(assaySampleRefs));
     }
 
+    @Override
+    public void insertUploadedFiles(SubmissionEnvelope submissionEnvelope) {
+        String submissionId = submissionEnvelope.getSubmission().getId();
+
+        List<UploadedFile> uploadedFiles = submissionEnvelope.getAssayData().stream().map(
+                AssayData::getFiles
+        ).collect(Collectors.toList())
+            .stream().flatMap(List::stream).collect(Collectors.toList())
```

Comment:
do we need to collect to a list here - can we carry on working on the stream?

Reply:
Agreed

Reply:
Removed